node/libs/protobuf_build/src/lib.rs

```diff
@@ -125,11 +127,14 @@ pub struct Config {
     pub input_root: InputPath,
     /// Implicit prefix that should be prepended to proto paths of the proto files in the input directory.
     pub proto_root: ProtoPath,
-    /// Descriptors of the dependencies and the rust absolute paths under which they will be available from the generated code.
-    pub dependencies: Vec<(RustName, &'static Descriptor)>,
-    /// Rust absolute path under which the protobuf crate will be available from the generated
-    /// code.
+    /// Descriptors of the direct dependencies and the rust absolute paths under which they will be available from the generated code.
+    /// Each dependency must be a direct dependency of the built crate. OTOH, it doesn't need to be a build dependency.
+    pub dependencies: Vec<RustName>,
```

Comment:
this means that from now on the proto module embedding the generated code will have to be publicly available (it cannot be private and reexported). Is that right? Can we do sth about that?

Reply:
I'm not sure I understand the question. The location of the original proto module *in the crate* doesn't matter for this purpose; what matters is that descriptors are publicly available (same as previously), and that they are available from the same crate they are generated for. The latter can be restricting; e.g. if `crate_b` re-exports `crate_a` in its entirety and we want to specify the dependency on it as `::crate_b::crate_a::proto`, it won't work. `crate_a` needs to be a direct dependency for `DEP_` env variables to be passed properly. Personally, this doesn't look like a deal breaker (`crate_a` is in the dependency graph of the target crate anyway), but it *is* a limitation.

Reply:
yes, that's the limitation I'm referring to. Also if you reexport crate_a::submodule::proto as crate_a::proto it also won't work.

Reply:
I'm not saying it's a deal breaker, I'm saying that it is a regression.

Reply:
> Also if you reexport crate_a::submodule::proto as crate_a::proto it also won't work.

It should work. I've tested locally that `zksync_protobuf::proto` works if re-exported as `zksync_protobuf::proto3`.